encodings/fastlanes/src/rle/mod.rs

```diff
@@ -162,59 +160,57 @@ impl RLEArray {
             dtype,
             values,
             indices,
-            value_chunk_offsets,
-            validity,
+            values_idx_offsets,
             stats_set: ArrayStats::default(),
             offset,
             length,
         }
     }
 
-    pub(crate) fn values(&self) -> &dyn Array {
+    #[inline]
+    pub fn values(&self) -> &ArrayRef {
         &self.values
     }
 
-    pub(crate) fn indices(&self) -> &dyn Array {
+    #[inline]
+    pub fn indices(&self) -> &ArrayRef {
         &self.indices
     }
 
-    pub(crate) fn value_chunk_offsets(&self) -> &dyn Array {
-        &self.value_chunk_offsets
-    }
-
-    pub fn ptype(&self) -> PType {
-        self.dtype.as_ptype()
+    #[inline]
+    pub fn values_idx_offsets(&self) -> &ArrayRef {
+        &self.values_idx_offsets
     }
 
-    /// Returns the offset within a chunk for an absolute position.
-    pub(crate) fn offset_in_chunk(&self, abs_position: usize) -> usize {
-        abs_position & (FL_CHUNK_SIZE - 1) // Equivalent to % 1024
-    }
-
-    /// Start index offset in the values array for a given chunk.
+    /// Values index offset relative to the first chunk.
+    ///
+    /// Offsets in the `values_idx_offsets` arrays are absolute and need to be
+    /// shifted by the offset of the first chunk, respective the current slice,
+    /// in order to make them relative.
     #[allow(clippy::expect_used)]
-    pub(crate) fn value_chunk_offset(&self, chunk_idx: usize) -> usize {
-        self.value_chunk_offsets
+    pub(crate) fn values_idx_offset(&self, chunk_idx: usize) -> usize {
```

Comment:
I think this method is fine as a one off but in decompression you're using it in the loop. In that case you should canonicalize the value_idx_offsets instead of going through scalar at